mmdet3d/models/dense_heads/fcaf3d_head.py

```diff
@@ -0,0 +1,682 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+# Adapted from https://github.com/SamsungLabs/fcaf3d/blob/master/mmdet3d/models/dense_heads/fcaf3d_neck_with_head.py # noqa
+try:
+    import MinkowskiEngine as ME
+except ImportError:
+    import warnings
+    warnings.warn(
+        'Please follow `getting_started.md` to install MinkowskiEngine.`')
+
+import torch
+from mmcv.cnn import Scale, bias_init_with_prob
+from mmcv.ops import nms3d, nms3d_normal
+from mmcv.runner.base_module import BaseModule
+from torch import nn
+
+from mmdet3d.core.bbox.structures import rotation_3d_in_axis
+from mmdet3d.models import HEADS, build_loss
+from mmdet.core import reduce_mean
+
+
+@HEADS.register_module()
+class FCAF3DHead(BaseModule):
+    """Bbox head of `FCAF3D <https://arxiv.org/abs/2112.00322>`_. Actually here
+    we store both the sparse 3D FPN and a head. The neck and the head can not
+    be simply separated as pruning score on the i-th level of FPN requires
+    classification scores from i+1-th level of the head.
+
+    Args:
+        n_classes (int): Number of classes.
+        in_channels (tuple[int]): Number of channels in input tensors.
+        out_channels (int): Number of channels in the neck output tensors.
+        n_reg_outs (int): Number of regression layer channels.
+        voxel_size (float): Voxel size in meters.
+        pts_prune_threshold (int): Pruning threshold on each feature level.
+        pts_assign_threshold (int): Min number of location per box to
+            be assigned with.
+        pts_center_threshold (int): Max number of locations per box to
+            be assigned with.
```

Comment:
May consider more clear names for the above two variables?

Reply:
Just added more documentation.